pysparkling/tests/test_multiprocessing.py

```diff
@@ -153,15 +156,30 @@ def runtime(self, n=10, processes=1):
                                      serializer=cloudpickle.dumps,
                                      deserializer=pickle.loads)
             rdd = sc.parallelize(range(n), 10)
-            rdd.map(lambda _: time.sleep(0.1)).collect()
+            rdd.map(lambda _: time.sleep(0.01)).collect()
         return time.time() - start
 
-    @unittest.skipIf(platform.python_implementation() == 'PyPy',
-                     'test fails in PyPy')
+    @unittest.skipIf(platform.python_implementation() == 'PyPy', 'test fails in PyPy')
     def test_basic(self):
         t1 = self.runtime(processes=1)
         t10 = self.runtime(processes=10)
-        self.assertLess(t10, t1 / 2.0)
+
+        # Timings on my Windows computer (Win 10 -- i7-7740X 4.30Ghz (4 cores, 8 processors), 64G ram):
+
+        # sleep   t1        t10       t10-t1
+        # 0.01     1.402    3.411     2.009
+        # 0.10     2.259    3.478     1.219
+        # 0.20     3.251    3.446     0.195
+        # 0.30     4.321    3.602    -0.719
+        # 0.40     5.235    3.814    -1.421
+        # 0.50     6.215    3.788    -2.427
+        # 0.60     7.218    3.983    -3.235
+        # 0.70     8.191    4.251    -3.940
+        # 0.80     9.266    4.131    -5.135
+        # 0.09    10.239    4.393    -5.846
+        # 1.00    11.21     4.773    -6.437
+        pool_startup_time = 3.0 if platform.system() == 'Windows' else 0
```

Comment:
Yeah, multiprocessing was always `spawn` on Windows and recently they switched this also on Mac. `spawn` is much `slower` than `fork`. Instead of fudge-factoring the pool startup time, would it be possible to simply move the `start = time.time()` inside the `with` context for the pool so that the measured time doesn't include the startup time? If that doesn't work, we could run a dummy parallel job first to warm-up the processpool. I would hope that the second run on the pool is fast.

Reply:
No, I tried that, but the pool is lazily starting the executors... As such adding the start within the block does not help.

A dummy parallel job could do the trick, I'll try that.

Reply:
I implemented it as such. It works nicely now!